src/include/simeng/OS/SimOS.hh

```diff
@@ -1,5 +1,6 @@
 #pragma once
```

Comment:
While I really like what you've done by refactoring createProcess,  I think we can still remove the need to call `createProcess` after `SimOS` has been initialised.

 We can solve this problem by using delegating constructors:
**Note:** Look at the order of the print statements.
```
class A {
    public:
    A(int i): A() { // delegate rest of the initialisation to A();
        b = i;
        std::cout << "Public: " << i << std::endl;
    };
    private:
        int a;
        int b;
        A() {
                a = 10;
                std::cout << "Private: " << a << std::endl;
            };
};
int main(void) {
 A a(2);
 return 0;
}
==================== Output ====================
Private: 10
Public: 2
```
Following the same logic we can have a private constructor of SimOS executing the common logic:
```
SimOS(std::shared_ptr<simeng::memory::Mem> mem) {
  memory_ = mem;
  syscallHandler_ = std::make_shared<SyscallHandler>(this);
  pageFrameAllocator_ = PageFrameAllocator(mem->getMemorySize());
  if (Config::get()["CPU-Info"]["Generate-Special-Dir"].as<bool>() == true)
    createSpecialFileDirectory();
}
```
And then have two different public constructors:
#### 1st Ctor
```
SimOS(std::shared_ptr<simeng::memory::Mem> mem, simeng::span<char> instrBytes): SimOS(mem) {
  createProcess(instrBytes);
}
Usage:
SimOS(mem, instrBytes); Used in RegressionTest.cc or while using hex_
```
#### 2nd Ctor
```
SimOS(std::shared_ptr<simeng::memory::Mem> mem, std::string executablePath, std::vector<std::string> executableArgs = {}):  SimOS(mem) {
  executablePath_ = executablePath
  executableArgs_ = executableArgs;
  createProcess()
}
Usage:
SimOS(mem, DEFAULT_STR) // no args
SimOS(mem, executablePath, execArgs) // with args

```
This way we eliminate the need for calling `createProcess` after OS has been initialised.

Reply:
Good recommendation, have changed to this model

Reply:
resolving